lib/matplotlib/pyplot.py

```diff
@@ -449,19 +449,25 @@ def figure(num=None,  # autoincrement if None, else integer from 1-N
            **kwargs
            ):
     """
-    Create a new figure.
+    Create a new figure, or activate an existing figure.
 
     Parameters
     ----------
     num : int or str, optional
-        If not provided, a new figure will be created, and the figure number
-        will be incremented. The figure objects holds this number in a `number`
-        attribute.
-        If num is provided, and a figure with this id already exists, make
-        it active, and returns a reference to it. If this figure does not
-        exists, create it and returns it.
-        If num is a string, the window title will be set to this figure's
-        *num*.
+        A unique identifier for the figure.
+
+        If a figure with that identifier already exists, this figure is made
+        active and returned.
+
+        If there is no figure with the identifier or *num* is not given, a new
+        figure is created, made active and returned. If *num* is not given,
+        an auto-generated integer value is used (starting at 1 and incremented
```

Comment:
the autogen'd integer value is also used when num is a str.

Reply:
Indeed that is true...

Reply:
:roll_eyes: I have no idea how to reasonable express this without sounding really silly. :sad: Help is welcome.

It appears that figures have a number and a label (which seems redundant to me - a single id would be enough). If *num* is a str, it is used as the label.

Reply:
The easiest would be to deprecate allowing to pass strings here :)

Reply:
Passing strings is super useful though (disclosure, I rely on this feature in by day-job) because it propagates through to the window title.   We have code that generates a bunch of figures for the user and gives them window titles like `motor_x vs detector_y` and such.

Reply:
It also propagates to the defaults suggestion via the 'save' button.

Reply:
It would be reasonable to move this functionality to e.g. `figure(label="foo")` (which doesn't work right now)...
(Note that right now if you do `figure("foo")` then `fig.get_label() == "foo"` (I don't think the label of a figure is used anywhere else), but you can't set back the title using `fig.set_label`; I guess it would make sense to make that (label == title) work?) (yes, not all figures have an owning window, we don't have to make that work in that case)